### ISSUE TITLE
Error handling deviation

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/python-jsonpreprocessor#219]
JSON code:

```
"keyP"   : "A",
"dictP"  : {"A" : 1, "B" : 2},
"newparam" : "${dictP['${indexP}']}"
```

Result:

```
'Invalid syntax! One or more than one opened or closed curly bracket is missing in expression '${dictP['${indexP}']}'.
           Please check the configuration file of the executed test!'
```

Yes, the syntax is invalid. But not because of missing brackets. There are no brackets missing. 

The syntax violation is the index placed within the curly brackets (instead of outside).

Same expression moved to the left hand side of the colon:

`"${dictP['${indexP}']}" : "newvalue"`

Result:

`'Invalid syntax: Found index or sub-element inside curly brackets in the parameter '${dictP['str(${indexP})']}''`

Now the root cause of the error is detected. But the error message contains a type cast (`str(${indexP})`) that is not part of the expression.

It is interesting to see, that the error handling is so much different depending on the position of the invalid expression.

It would be better to have the behavior aligned.

**Comment by namsonx** [comment]
Hello Holger,

This error handling deviation was fixed while doing improvement. The current status is:

For the case
```
"keyP"   : "A",
"dictP"  : {"A" : 1, "B" : 2},
"newparam" : "${dictP['${indexP}']}"
```
-> Error message: ```Exception: Invalid syntax: Found index or sub-element inside curly brackets in the parameter '${dictP['${indexP}']}'```

In case the same expression moved to the left hand side of the colon:
```
"keyP"   : "A",
"dictP"  : {"A" : 1, "B" : 2},
"${dictP['${indexP}']}" : "newvalue"
```
-> It shows ```Exception: A substitution in key names is not allowed! Please update the key name "${dictP['${indexP}']}"``` because JPP does not allow a substitution in key names.

If users make this mistake like below:
```
"keyP"   : "A",
"dictP"  : {"A" : 1, "B" : 2},
${dictP['${indexP}']} : "newvalue"
```
-> The same error message as the first case is shown: ```Exception: Invalid syntax: Found index or sub-element inside curly brackets in the parameter '${dictP['${indexP}']}'```

Could you please help me verify this ticket again?

Thank you,
Son

**Comment by HolQue** [comment]
Retest successful. Issue can be closed.

**Comment by test-fullautomation** [comment]
released with RobotFramework AIO 0.13.0
